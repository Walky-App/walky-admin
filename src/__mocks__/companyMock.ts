import { type ICompany } from '../interfaces/company'

export const companyMock: ICompany = {
  _id: '66ccbb8513c98d0f7012a44b',
  company_location_pin: [37.6782534, -120.9941554],
  company_name: 'Fake company for mocking purposes',
  company_address: '2608 McHenry Avenue, Modesto, CA, USA',
  company_city: 'Modesto',
  company_country: 'United States',
  company_dbas: ['Dank Flowers'],
  company_phone_number: '2223334444',
  company_state: 'CA',
  company_tax_id: '22-3333333',
  company_zip: '95350',
  created_by: 'donnie-client-2@hemptemps.com',
  facilities: ['66ccbd1513c98d0f7012a8f4'],
  users: ['66843da2c853cc8e1a1fc290'],
  createdAt: '2024-08-26T17:29:41.802Z',
  company_documents: [],
  payment_information: [],
  updatedAt: '2024-08-26T17:36:22.065Z',
}
