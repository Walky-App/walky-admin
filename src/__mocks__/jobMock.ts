import { type IUser } from '../interfaces/User'
import { type IFacility } from '../interfaces/facility'
import { type IJob, type IJobShiftDay, type IRestriction, type IFeedback, type IDnr } from '../interfaces/job'

export const jobMock: IJob = {
  _id: '66bf8bc2907a582955d1b0c1',
  uid: '2408711',
  created_by: 'shannon@hemptemps.com',
  facility: {} as IFacility,
  title: 'Trimmer',
  start_time: '2024-08-16T12:30:00.000+00:00',
  end_time: '2024-08-16T21:00:00.000+00:00',
  total_hours: 8,
  lunch_break: 30,
  job_days: [] as IJobShiftDay[],
  job_dates: [
    '2024-08-19T06:00:00.000+00:00',
    '2024-08-20T06:00:00.000+00:00',
    '2024-08-21T06:00:00.000+00:00',
    '2024-08-22T06:00:00.000+00:00',
    '2024-08-23T06:00:00.000+00:00',
  ],
  job_tips: ['Parking Onsite', 'Parking on Street'],
  vacancy: 10,
  hourly_rate: 19,
  applicants: [
    {
      user: {} as IUser,
      is_active: false,
      is_working: false,
      _id: '66bf9e8f2fe241f8e7b433e7',
      first_name: 'John',
      last_name: 'Doe',
      rejection_reason: '',
    },
    {
      user: {} as IUser,
      is_active: false,
      is_working: false,
      _id: '66bf9f012fe241f8e7b43edd',
      first_name: 'Jane',
      last_name: 'Doe',
      rejection_reason: 'Some weird reason',
    },
  ],
  saved_by: [],
  dnr: [] as IDnr[],
  is_completed: true,
  is_full: false,
  is_active: true,
  feedback: [] as IFeedback[],
  clock_in_restriction: [] as IRestriction[],
  clock_out_restriction: [] as IRestriction[],
  distance: 0,
  createdAt: new Date('2024-08-16T17:26:26.729+00:00'),
  updatedAt: new Date('2024-08-24T06:00:09.550+00:00'),
  applicants_feedback_ids: [],
}
