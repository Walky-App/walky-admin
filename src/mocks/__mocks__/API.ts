// __mocks__/API.ts
const API = {
    post: jest.fn(), // we'll customize this in individual tests
  };
  
  export default API;
  