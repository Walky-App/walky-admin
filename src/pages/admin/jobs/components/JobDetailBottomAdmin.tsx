import { TabPanel, TabView } from 'primereact/tabview'

import { type IJob } from '../../../../interfaces/job'
import { DroppedShiftsByJob } from './DroppedShiftsByJob'
import { JobDetailApplicantTimesheetTabAdmin } from './JobDetailApplicantTimesheetTabAdmin'

export const JobDetailBottomAdmin = ({ job }: { job: IJob }) => {
  return (
    <div className="col-span-1 md:col-span-3">
      <TabView className="mt-4">
        <TabPanel header="Timesheet">
          <JobDetailApplicantTimesheetTabAdmin job={job} />
        </TabPanel>
        <TabPanel header="Facility Images">{facilityImagesTemplate(job)}</TabPanel>
        <TabPanel header="Dropped Shifts">
          <DroppedShiftsByJob jobId={job._id} />
        </TabPanel>
      </TabView>
    </div>
  )
}

const facilityImagesTemplate = (job: IJob) => {
  return (
    <section className="mt-4">
      <h2 className="text-base font-semibold leading-6">Facility Images</h2>
      <ul className="grid grid-cols-2 gap-x-4 gap-y-8 sm:grid-cols-3 sm:gap-x-6 lg:grid-cols-4 xl:gap-x-8">
        {job?.facility?.images?.map(image => (
          <li key={image._id} className="relative">
            <div className="aspect-h-7 aspect-w-10 group block w-full overflow-hidden rounded-lg bg-gray-100 focus-within:ring-2 focus-within:ring-green-500 focus-within:ring-offset-2 focus-within:ring-offset-gray-100">
              <img src={image.url} alt="" className="pointer-events-none h-80 object-cover group-hover:opacity-75" />
            </div>
          </li>
        ))}
      </ul>
    </section>
  )
}
