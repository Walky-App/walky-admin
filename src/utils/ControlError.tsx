import type { FieldErrors } from 'react-hook-form'

export function getFormErrorMessage(path: string, errors: FieldErrors) {
  const pathParts = path.split('.')
  let error: FieldErrors = errors

  for (const part of pathParts) {
    if (typeof error !== 'object' || error === null) {
      return null
    }
    error = error[part as keyof typeof error] as FieldErrors
  }

  if (error?.message) {
    return error.message ? <p className="mt-2 text-sm text-red-600">{String(error.message)}</p> : null
  }

  return null
}
