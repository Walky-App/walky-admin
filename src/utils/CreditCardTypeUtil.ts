export const getCardType = (number: string) => {
  const firstDigit = number[0]
  const firstTwoDigits = number.slice(0, 2)
  const firstFourDigits = number.slice(0, 4)

  if (/^4/.test(firstDigit)) {
    return 'Visa'
  } else if (/^5[1-5]/.test(firstTwoDigits)) {
    return 'MasterCard'
  } else if (/^3[47]/.test(firstTwoDigits)) {
    return 'Amex'
  } else if (/^6(?:011|5)/.test(firstFourDigits)) {
    return 'Discover'
  } else {
    return 'Unknown'
  }
}
