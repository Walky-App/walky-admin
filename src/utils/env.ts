export const getEnv = () => {
  return {
    VITE_API_BASE_URL: import.meta.env.VITE_API_BASE_URL || 'https://staging.walkyapp.com/api'
  };
};
