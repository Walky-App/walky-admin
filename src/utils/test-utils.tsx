import React, { type ReactElement } from 'react'

import { MemoryRouter } from 'react-router-dom'

import '@testing-library/jest-dom'
import { render, type RenderOptions } from '@testing-library/react'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  const originalConsoleError = console.error
  const jsDomCssError = 'Error: Could not parse CSS stylesheet'
  console.error = (...params) => {
    if (!params.find(p => p.toString().includes(jsDomCssError))) {
      originalConsoleError(...params)
    }
  }
  return <MemoryRouter>{children}</MemoryRouter>
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
