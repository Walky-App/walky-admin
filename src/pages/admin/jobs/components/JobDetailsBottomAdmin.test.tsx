import { jobMock } from '../../../../__mocks__/jobMock'
import { render } from '../../../../utils/test-utils'
import { JobDetailBottomAdmin } from './JobDetailBottomAdmin'

describe('JobDetailBottomAdmin', () => {
  it('should render job detail bottom admin component', () => {
    render(<JobDetailBottomAdmin job={jobMock} />)
  })

  it('should match snapshot', () => {
    const { asFragment } = render(<JobDetailBottomAdmin job={jobMock} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
