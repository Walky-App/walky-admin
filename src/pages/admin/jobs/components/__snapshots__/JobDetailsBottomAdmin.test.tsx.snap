// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobDetailBottomAdmin should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="col-span-1 md:col-span-3"
  >
    <div
      class="mt-4 p-tabview p-component"
      data-pc-name="tabview"
      data-pc-section="root"
      id="pr_id_4"
    >
      <div
        class="p-tabview-nav-container"
        data-pc-section="navcontainer"
      >
        <div
          class="p-tabview-nav-content"
          data-pc-section="navcontent"
          id="pr_id_4_navcontent"
        >
          <ul
            class="p-tabview-nav"
            data-pc-section="nav"
            role="tablist"
          >
            <li
              class="p-unselectable-text p-tabview-selected p-highlight"
              data-pc-name=""
              data-pc-section="header"
              role="presentation"
            >
              <a
                aria-controls="pr_id_40_content"
                aria-disabled="false"
                aria-selected="true"
                class="p-tabview-nav-link"
                data-pc-section="headeraction"
                id="pr_id_4_header_0"
                role="tab"
                tabindex="0"
              >
                <span
                  class="p-tabview-title"
                  data-pc-section="headertitle"
                >
                  Timesheet
                </span>
              </a>
            </li>
            <li
              class="p-unselectable-text"
              data-pc-name=""
              data-pc-section="header"
              role="presentation"
            >
              <a
                aria-controls="pr_id_41_content"
                aria-disabled="false"
                aria-selected="false"
                class="p-tabview-nav-link"
                data-pc-section="headeraction"
                id="pr_id_4_header_1"
                role="tab"
                tabindex="-1"
              >
                <span
                  class="p-tabview-title"
                  data-pc-section="headertitle"
                >
                  Facility Images
                </span>
              </a>
            </li>
            <li
              class="p-unselectable-text"
              data-pc-name=""
              data-pc-section="header"
              role="presentation"
            >
              <a
                aria-controls="pr_id_42_content"
                aria-disabled="false"
                aria-selected="false"
                class="p-tabview-nav-link"
                data-pc-section="headeraction"
                id="pr_id_4_header_2"
                role="tab"
                tabindex="-1"
              >
                <span
                  class="p-tabview-title"
                  data-pc-section="headertitle"
                >
                  Dropped Shifts
                </span>
              </a>
            </li>
            <li
              aria-hidden="true"
              class="p-tabview-ink-bar"
              data-pc-section="inkbar"
              role="presentation"
              style="left: 0px;"
            />
          </ul>
        </div>
      </div>
      <div
        class="p-tabview-panels"
        data-pc-section="panelcontainer"
      >
        <div
          aria-labelledby="pr_id_4_header_0"
          class="p-tabview-panel"
          data-pc-name=""
          data-pc-section="content"
          id="pr_id_40_content"
          role="tabpanel"
        >
          <section
            class="space-y-6"
          >
            <div
              class="mt-4 items-end justify-between space-y-4 sm:flex"
            >
              <div>
                <div
                  class="flex flex-row items-center gap-x-2"
                >
                  <i
                    class="custom-target-icon pi pi-info-circle p-text-secondary p-overlay-badge cursor-pointer text-lg lg:text-sm"
                    data-pr-at="left top-5"
                    data-pr-classname="text-sm text-wrap"
                    data-pr-my="left bottom"
                    data-pr-position="top"
                    data-pr-tooltip="Select an applicant to view their timesheet"
                  />
                  <label
                    class="block text-sm font-medium leading-6"
                    for="applicant_dropdown"
                  >
                    Applicant:
                  </label>
                </div>
                <div
                  class="p-inputgroup"
                >
                  <div
                    class="p-dropdown p-component p-inputwrapper"
                    data-p-disabled="false"
                    data-p-focus="false"
                    data-pc-name="dropdown"
                    data-pc-section="root"
                  >
                    <div
                      class="p-hidden-accessible"
                      data-pc-section="hiddenselectedmessage"
                    >
                      <input
                        aria-haspopup="listbox"
                        data-pc-section="input"
                        readonly=""
                        tabindex="0"
                        type="text"
                        value=""
                      />
                    </div>
                    <div
                      class="p-hidden-accessible p-dropdown-hidden-select"
                      data-pc-section="hiddenselectedmessage"
                    >
                      <select
                        aria-hidden="true"
                        data-pc-section="select"
                        name="applicant_dropdown"
                        tabindex="-1"
                      >
                        <option
                          data-pc-section="option"
                          selected=""
                          value=""
                        >
                          Select an applicant
                        </option>
                      </select>
                    </div>
                    <span
                      class="p-dropdown-label p-inputtext p-placeholder"
                      data-pc-section="input"
                      tabindex="-1"
                    >
                      Select an applicant
                    </span>
                    <div
                      aria-expanded="false"
                      aria-haspopup="listbox"
                      aria-label="Select an applicant"
                      class="p-dropdown-trigger"
                      data-pc-section="trigger"
                      role="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="p-icon p-dropdown-trigger-icon p-clickable"
                        data-pc-section="dropdownicon"
                        data-pr-overlay-visible="false"
                        fill="none"
                        height="14"
                        viewBox="0 0 14 14"
                        width="14"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M7.01744 10.398C6.91269 10.3985 6.8089 10.378 6.71215 10.3379C6.61541 10.2977 6.52766 10.2386 6.45405 10.1641L1.13907 4.84913C1.03306 4.69404 0.985221 4.5065 1.00399 4.31958C1.02276 4.13266 1.10693 3.95838 1.24166 3.82747C1.37639 3.69655 1.55301 3.61742 1.74039 3.60402C1.92777 3.59062 2.11386 3.64382 2.26584 3.75424L7.01744 8.47394L11.769 3.75424C11.9189 3.65709 12.097 3.61306 12.2748 3.62921C12.4527 3.64535 12.6199 3.72073 12.7498 3.84328C12.8797 3.96582 12.9647 4.12842 12.9912 4.30502C13.0177 4.48162 12.9841 4.662 12.8958 4.81724L7.58083 10.1322C7.50996 10.2125 7.42344 10.2775 7.32656 10.3232C7.22968 10.3689 7.12449 10.3944 7.01744 10.398Z"
                          fill="currentColor"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
              <div>
                <div
                  class="flex flex-row items-center gap-x-2"
                >
                  <i
                    class="custom-target-icon pi pi-info-circle p-text-secondary p-overlay-badge cursor-pointer text-lg lg:text-sm"
                    data-pr-at="left top-5"
                    data-pr-classname="text-sm text-wrap"
                    data-pr-my="left bottom"
                    data-pr-position="top"
                    data-pr-tooltip="Select a shift to view and/or clock in/out for the selected applicant"
                  />
                  <label
                    class="block text-sm font-medium leading-6"
                    for="applicant_shift_dropdown"
                  >
                    Shift:
                  </label>
                </div>
                <div
                  class="p-inputgroup"
                >
                  <button
                    aria-label="Reset"
                    class="p-button p-component p-disabled p-button-secondary"
                    data-pc-name="button"
                    data-pc-section="root"
                    disabled=""
                  >
                    <span
                      class="p-button-icon p-c p-button-icon-left pi pi-refresh"
                      data-pc-section="icon"
                    />
                    <span
                      class="p-button-label p-c"
                      data-pc-section="label"
                    >
                      Reset
                    </span>
                  </button>
                  <div
                    class="p-dropdown p-component p-inputwrapper p-disabled"
                    data-p-disabled="true"
                    data-p-focus="false"
                    data-pc-name="dropdown"
                    data-pc-section="root"
                  >
                    <div
                      class="p-hidden-accessible"
                      data-pc-section="hiddenselectedmessage"
                    >
                      <input
                        aria-haspopup="listbox"
                        data-pc-section="input"
                        disabled=""
                        readonly=""
                        tabindex="-1"
                        type="text"
                        value=""
                      />
                    </div>
                    <div
                      class="p-hidden-accessible p-dropdown-hidden-select"
                      data-pc-section="hiddenselectedmessage"
                    >
                      <select
                        aria-hidden="true"
                        data-pc-section="select"
                        name="applicant_shift_dropdown"
                        tabindex="-1"
                      >
                        <option
                          data-pc-section="option"
                          selected=""
                          value=""
                        >
                          Select a shift
                        </option>
                      </select>
                    </div>
                    <span
                      class="p-dropdown-label p-inputtext p-placeholder"
                      data-pc-section="input"
                      tabindex="-1"
                    >
                      Select a shift
                    </span>
                    <div
                      aria-expanded="false"
                      aria-haspopup="listbox"
                      aria-label="Select a shift"
                      class="p-dropdown-trigger"
                      data-pc-section="trigger"
                      role="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="p-icon p-dropdown-trigger-icon p-clickable"
                        data-pc-section="dropdownicon"
                        data-pr-overlay-visible="false"
                        fill="none"
                        height="14"
                        viewBox="0 0 14 14"
                        width="14"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M7.01744 10.398C6.91269 10.3985 6.8089 10.378 6.71215 10.3379C6.61541 10.2977 6.52766 10.2386 6.45405 10.1641L1.13907 4.84913C1.03306 4.69404 0.985221 4.5065 1.00399 4.31958C1.02276 4.13266 1.10693 3.95838 1.24166 3.82747C1.37639 3.69655 1.55301 3.61742 1.74039 3.60402C1.92777 3.59062 2.11386 3.64382 2.26584 3.75424L7.01744 8.47394L11.769 3.75424C11.9189 3.65709 12.097 3.61306 12.2748 3.62921C12.4527 3.64535 12.6199 3.72073 12.7498 3.84328C12.8797 3.96582 12.9647 4.12842 12.9912 4.30502C13.0177 4.48162 12.9841 4.662 12.8958 4.81724L7.58083 10.1322C7.50996 10.2125 7.42344 10.2775 7.32656 10.3232C7.22968 10.3689 7.12449 10.3944 7.01744 10.398Z"
                          fill="currentColor"
                        />
                      </svg>
                    </div>
                  </div>
                  <button
                    aria-label="Clock-In For Applicant"
                    class="p-button p-component p-disabled"
                    data-pc-name="button"
                    data-pc-section="root"
                    disabled=""
                  >
                    <span
                      class="p-button-label p-c text-nowrap"
                      data-pc-section="label"
                    >
                      Clock-In For Applicant
                    </span>
                  </button>
                </div>
              </div>
            </div>
            <div>
              <h2
                class="text-base font-semibold leading-6"
              >
                Timesheet
              </h2>
              <div
                class="p-datatable p-component p-datatable-responsive-scroll p-datatable-striped"
                data-pc-name="datatable"
                data-pc-section="root"
                data-scrollselectors=".p-datatable-wrapper"
                data-showgridlines="false"
                pr_id_3=""
              >
                <div
                  class="p-datatable-wrapper"
                  data-pc-section="wrapper"
                  style="overflow: auto;"
                >
                  <table
                    class="p-datatable-table"
                    data-pc-section="table"
                    role="table"
                  >
                    <thead
                      class="p-datatable-thead"
                      data-pc-name="datatable"
                      data-pc-section="thead"
                      role="rowgroup"
                    >
                      <tr
                        data-pc-section="headerrow"
                        role="row"
                      >
                        <th
                          data-p-filter-column="false"
                          data-p-frozen-column="false"
                          data-p-reorderable-column="false"
                          data-p-resizable-column="false"
                          data-p-sortable-column="false"
                          data-pc-name="datatable"
                          data-pc-section="headercell"
                          role="columnheader"
                        >
                          <div
                            class="p-column-header-content"
                            data-pc-section="headercontent"
                          >
                            <span
                              class="p-column-title"
                              data-pc-section="headertitle"
                            >
                              Shift Day
                            </span>
                          </div>
                        </th>
                        <th
                          data-p-filter-column="false"
                          data-p-frozen-column="false"
                          data-p-reorderable-column="false"
                          data-p-resizable-column="false"
                          data-p-sortable-column="false"
                          data-pc-name="datatable"
                          data-pc-section="headercell"
                          role="columnheader"
                        >
                          <div
                            class="p-column-header-content"
                            data-pc-section="headercontent"
                          >
                            <span
                              class="p-column-title"
                              data-pc-section="headertitle"
                            >
                              Time In
                            </span>
                          </div>
                        </th>
                        <th
                          data-p-filter-column="false"
                          data-p-frozen-column="false"
                          data-p-reorderable-column="false"
                          data-p-resizable-column="false"
                          data-p-sortable-column="false"
                          data-pc-name="datatable"
                          data-pc-section="headercell"
                          role="columnheader"
                        >
                          <div
                            class="p-column-header-content"
                            data-pc-section="headercontent"
                          >
                            <span
                              class="p-column-title"
                              data-pc-section="headertitle"
                            >
                              Time Out
                            </span>
                          </div>
                        </th>
                        <th
                          data-p-filter-column="false"
                          data-p-frozen-column="false"
                          data-p-reorderable-column="false"
                          data-p-resizable-column="false"
                          data-p-sortable-column="false"
                          data-pc-name="datatable"
                          data-pc-section="headercell"
                          role="columnheader"
                        >
                          <div
                            class="p-column-header-content"
                            data-pc-section="headercontent"
                          >
                            <span
                              class="p-column-title"
                              data-pc-section="headertitle"
                            >
                              Lunch Break
                            </span>
                          </div>
                        </th>
                        <th
                          data-p-filter-column="false"
                          data-p-frozen-column="false"
                          data-p-reorderable-column="false"
                          data-p-resizable-column="false"
                          data-p-sortable-column="false"
                          data-pc-name="datatable"
                          data-pc-section="headercell"
                          role="columnheader"
                        >
                          <div
                            class="p-column-header-content"
                            data-pc-section="headercontent"
                          >
                            <span
                              class="p-column-title"
                              data-pc-section="headertitle"
                            >
                              Scheduled Hours
                            </span>
                          </div>
                        </th>
                        <th
                          data-p-filter-column="false"
                          data-p-frozen-column="false"
                          data-p-reorderable-column="false"
                          data-p-resizable-column="false"
                          data-p-sortable-column="false"
                          data-pc-name="datatable"
                          data-pc-section="headercell"
                          role="columnheader"
                        >
                          <div
                            class="p-column-header-content"
                            data-pc-section="headercontent"
                          >
                            <span
                              class="p-column-title"
                              data-pc-section="headertitle"
                            >
                              Total Hours
                            </span>
                          </div>
                        </th>
                      </tr>
                    </thead>
                    <tbody
                      class="p-datatable-tbody"
                      data-pc-section="tbody"
                      role=" rowgroup"
                    >
                      <tr
                        class="p-datatable-emptymessage"
                        data-pc-section="emptymessage"
                        role="row"
                      >
                        <td
                          colspan="6"
                          data-pc-name="datatable"
                          data-pc-section="bodycell"
                          role="cell"
                        >
                          No time data to display
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
                <div
                  class="p-paginator-bottom p-paginator p-component"
                  data-pc-name="paginator"
                  data-pc-section="root"
                >
                  <button
                    aria-label="First Page"
                    class="p-paginator-first p-paginator-element p-link p-disabled"
                    data-pc-section="firstpagebutton"
                    disabled=""
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="p-icon p-paginator-icon"
                      data-pc-section="firstpageicon"
                      fill="none"
                      height="14"
                      viewBox="0 0 14 14"
                      width="14"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M5.71602 11.164C5.80782 11.2021 5.9063 11.2215 6.00569 11.221C6.20216 11.2301 6.39427 11.1612 6.54025 11.0294C6.68191 10.8875 6.76148 10.6953 6.76148 10.4948C6.76148 10.2943 6.68191 10.1021 6.54025 9.96024L3.51441 6.9344L6.54025 3.90855C6.624 3.76126 6.65587 3.59011 6.63076 3.42254C6.60564 3.25498 6.525 3.10069 6.40175 2.98442C6.2785 2.86815 6.11978 2.79662 5.95104 2.7813C5.78229 2.76598 5.61329 2.80776 5.47112 2.89994L1.97123 6.39983C1.82957 6.54167 1.75 6.73393 1.75 6.9344C1.75 7.13486 1.82957 7.32712 1.97123 7.46896L5.47112 10.9991C5.54096 11.0698 5.62422 11.1259 5.71602 11.164ZM11.0488 10.9689C11.1775 11.1156 11.3585 11.2061 11.5531 11.221C11.7477 11.2061 11.9288 11.1156 12.0574 10.9689C12.1815 10.8302 12.25 10.6506 12.25 10.4645C12.25 10.2785 12.1815 10.0989 12.0574 9.96024L9.03158 6.93439L12.0574 3.90855C12.1248 3.76739 12.1468 3.60881 12.1204 3.45463C12.0939 3.30045 12.0203 3.15826 11.9097 3.04765C11.7991 2.93703 11.6569 2.86343 11.5027 2.83698C11.3486 2.81053 11.19 2.83252 11.0488 2.89994L7.51865 6.36957C7.37699 6.51141 7.29742 6.70367 7.29742 6.90414C7.29742 7.1046 7.37699 7.29686 7.51865 7.4387L11.0488 10.9689Z"
                        fill="currentColor"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </button>
                  <button
                    aria-label="Previous Page"
                    class="p-paginator-prev p-paginator-element p-link p-disabled"
                    data-pc-section="prevpagebutton"
                    disabled=""
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="p-icon p-paginator-icon"
                      data-pc-section="prevpageicon"
                      fill="none"
                      height="14"
                      viewBox="0 0 14 14"
                      width="14"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M8.75 11.185C8.65146 11.1854 8.55381 11.1662 8.4628 11.1284C8.37179 11.0906 8.28924 11.0351 8.22 10.965L4.72 7.46496C4.57955 7.32433 4.50066 7.13371 4.50066 6.93496C4.50066 6.73621 4.57955 6.54558 4.72 6.40496L8.22 2.93496C8.36095 2.84357 8.52851 2.80215 8.69582 2.81733C8.86312 2.83252 9.02048 2.90344 9.14268 3.01872C9.26487 3.134 9.34483 3.28696 9.36973 3.4531C9.39463 3.61924 9.36303 3.78892 9.28 3.93496L6.28 6.93496L9.28 9.93496C9.42045 10.0756 9.49934 10.2662 9.49934 10.465C9.49934 10.6637 9.42045 10.8543 9.28 10.995C9.13526 11.1257 8.9448 11.1939 8.75 11.185Z"
                        fill="currentColor"
                      />
                    </svg>
                  </button>
                  <span
                    class="p-paginator-pages"
                    data-pc-section="pages"
                  />
                  <button
                    aria-label="Next Page"
                    class="p-paginator-next p-paginator-element p-link p-disabled"
                    data-pc-section="nextpagebutton"
                    disabled=""
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="p-icon p-paginator-icon"
                      data-pc-section="nextpageicon"
                      fill="none"
                      height="14"
                      viewBox="0 0 14 14"
                      width="14"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M5.25 11.1728C5.14929 11.1694 5.05033 11.1455 4.9592 11.1025C4.86806 11.0595 4.78666 10.9984 4.72 10.9228C4.57955 10.7822 4.50066 10.5916 4.50066 10.3928C4.50066 10.1941 4.57955 10.0035 4.72 9.86283L7.72 6.86283L4.72 3.86283C4.66067 3.71882 4.64765 3.55991 4.68275 3.40816C4.71785 3.25642 4.79932 3.11936 4.91585 3.01602C5.03238 2.91268 5.17819 2.84819 5.33305 2.83149C5.4879 2.81479 5.64411 2.84671 5.78 2.92283L9.28 6.42283C9.42045 6.56346 9.49934 6.75408 9.49934 6.95283C9.49934 7.15158 9.42045 7.34221 9.28 7.48283L5.78 10.9228C5.71333 10.9984 5.63193 11.0595 5.5408 11.1025C5.44966 11.1455 5.35071 11.1694 5.25 11.1728Z"
                        fill="currentColor"
                      />
                    </svg>
                  </button>
                  <button
                    aria-label="Last Page"
                    class="p-paginator-last p-paginator-element p-link p-disabled"
                    data-pc-section="lastpagebutton"
                    disabled=""
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="p-icon p-paginator-icon"
                      data-pc-section="lastpageicon"
                      fill="none"
                      height="14"
                      viewBox="0 0 14 14"
                      width="14"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M7.68757 11.1451C7.7791 11.1831 7.8773 11.2024 7.9764 11.2019C8.07769 11.1985 8.17721 11.1745 8.26886 11.1312C8.36052 11.088 8.44238 11.0265 8.50943 10.9505L12.0294 7.49085C12.1707 7.34942 12.25 7.15771 12.25 6.95782C12.25 6.75794 12.1707 6.56622 12.0294 6.42479L8.50943 2.90479C8.37014 2.82159 8.20774 2.78551 8.04633 2.80192C7.88491 2.81833 7.73309 2.88635 7.6134 2.99588C7.4937 3.10541 7.41252 3.25061 7.38189 3.40994C7.35126 3.56927 7.37282 3.73423 7.44337 3.88033L10.4605 6.89748L7.44337 9.91463C7.30212 10.0561 7.22278 10.2478 7.22278 10.4477C7.22278 10.6475 7.30212 10.8393 7.44337 10.9807C7.51301 11.0512 7.59603 11.1071 7.68757 11.1451ZM1.94207 10.9505C2.07037 11.0968 2.25089 11.1871 2.44493 11.2019C2.63898 11.1871 2.81949 11.0968 2.94779 10.9505L6.46779 7.49085C6.60905 7.34942 6.68839 7.15771 6.68839 6.95782C6.68839 6.75793 6.60905 6.56622 6.46779 6.42479L2.94779 2.90479C2.80704 2.83757 2.6489 2.81563 2.49517 2.84201C2.34143 2.86839 2.19965 2.94178 2.08936 3.05207C1.97906 3.16237 1.90567 3.30415 1.8793 3.45788C1.85292 3.61162 1.87485 3.76975 1.94207 3.9105L4.95922 6.92765L1.94207 9.9448C1.81838 10.0831 1.75 10.2621 1.75 10.4477C1.75 10.6332 1.81838 10.8122 1.94207 10.9505Z"
                        fill="currentColor"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </button>
                  <div
                    class="p-dropdown p-component p-inputwrapper p-disabled p-inputwrapper-filled"
                    data-p-disabled="true"
                    data-p-focus="false"
                    data-pc-name="dropdown"
                    data-pc-section="root"
                  >
                    <div
                      class="p-hidden-accessible"
                      data-pc-section="hiddenselectedmessage"
                    >
                      <input
                        aria-haspopup="listbox"
                        aria-label="Jump to Page Dropdown"
                        data-pc-section="input"
                        disabled=""
                        readonly=""
                        tabindex="-1"
                        type="text"
                        value="7"
                      />
                    </div>
                    <div
                      class="p-hidden-accessible p-dropdown-hidden-select"
                      data-pc-section="hiddenselectedmessage"
                    >
                      <select
                        aria-hidden="true"
                        data-pc-section="select"
                        tabindex="-1"
                      >
                        <option
                          data-pc-section="option"
                          selected=""
                          value="7"
                        >
                          7
                        </option>
                      </select>
                    </div>
                    <span
                      class="p-dropdown-label p-inputtext"
                      data-pc-section="input"
                      tabindex="-1"
                    >
                      7
                    </span>
                    <div
                      aria-expanded="false"
                      aria-haspopup="listbox"
                      aria-label="Choose"
                      class="p-dropdown-trigger"
                      data-pc-section="trigger"
                      role="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="p-icon p-dropdown-trigger-icon p-clickable"
                        data-pc-section="dropdownicon"
                        data-pr-overlay-visible="false"
                        fill="none"
                        height="14"
                        viewBox="0 0 14 14"
                        width="14"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M7.01744 10.398C6.91269 10.3985 6.8089 10.378 6.71215 10.3379C6.61541 10.2977 6.52766 10.2386 6.45405 10.1641L1.13907 4.84913C1.03306 4.69404 0.985221 4.5065 1.00399 4.31958C1.02276 4.13266 1.10693 3.95838 1.24166 3.82747C1.37639 3.69655 1.55301 3.61742 1.74039 3.60402C1.92777 3.59062 2.11386 3.64382 2.26584 3.75424L7.01744 8.47394L11.769 3.75424C11.9189 3.65709 12.097 3.61306 12.2748 3.62921C12.4527 3.64535 12.6199 3.72073 12.7498 3.84328C12.8797 3.96582 12.9647 4.12842 12.9912 4.30502C13.0177 4.48162 12.9841 4.662 12.8958 4.81724L7.58083 10.1322C7.50996 10.2125 7.42344 10.2775 7.32656 10.3232C7.22968 10.3689 7.12449 10.3944 7.01744 10.398Z"
                          fill="currentColor"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </section>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
